src/datadog/trace_id.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+// This component provides a `struct`, `TraceID`, that represents an opaque,
+// unique identifier for a trace.
+// `TraceID` is 128 bits wide, though in some contexts only the lower 64 bits
+// are used.
+
+#include <cstdint>
+#include <string>
+
+#include "expected.h"
+#include "string_view.h"
+
+namespace datadog {
+namespace tracing {
+
+struct TraceID {
+  std::uint64_t low;
+  std::uint64_t high;
+
+  // Create a zero trace ID.
+  TraceID();
+
+  // Create a trace ID whose lower 64 bits are the specified `low` and whose
+  // higher 64 bits are zero.
+  explicit TraceID(std::uint64_t low);
+
+  // Create a trace ID whose lower 64 bits are the specified `low` and whose
+  // higher 64 bits are the specified `high`.
+  TraceID(std::uint64_t low, std::uint64_t high);
+
+  // Return a 32 character lower-case hexadecimal representation of this trace
+  // ID, padded with zeroes on the left.
+  std::string hex_padded() const;
+
+  // Return either a decimal or a "0x"-prefixed hexadecimal representation of
```

Comment:
Not sure I like this, a straight token is easier to lug around (copy-paste) if it's being used for error diagnostics.